draft-meynell-panrg-scion-research-questions.md

```diff
@@ -303,6 +333,41 @@ Do we actually need to solve this reverse path refresh problem?
   are anyway required.
 
 
+## Signalling faulty segments
+
+Faulty segments are segments that do not work as advertised, i.e. they are either physically faulty
+(broken link, high packet loss, jitter, ...) or come with faulty metadata, suggesting too few hops, too low latency,
+too much bandwith, or similar problems.
+The faultyness may be caused by a technical problem (e.g. broken link) or by a malicious adversary.
+
+An example for a malicious adversary is the "wormhole" attack, see {{I-D.scion-cp}}, where, an AS may be coaxed to
+disseminate a faulty segment. How do we recover from faulty segments?
+Currently only endpoints may realize that a segment does not work as advertised, either via SCMP errors or simply by traffic degradation.
+This would be a possible sequence of events:
+
+* An endpoint realizes that a segment is faulty
+* An endpoint needs to signal to its local AS that a segment is faulty
+* Local AS needs to signal its CORE AS that a segment is faulty
+* The CORE AS needs to:
+  * change policy to exclude faulty segments AND/OR
+  * signal other COREs and ISDs to stop delivering the segment (they only deliver 5 each, so any faulty segment shoud be avoided)
+
+### Implications
+
+A lack of recovery can lead to an attack as described in {{dos-with-wormholes}}.
+
+### Mitigation
+
+* An AS could monitor traffic for SCMP errors, however this works only if these errors are actually generated and forwarded.
+* Endpoints need a way to signal faulty segments to ASes. ASes need a way to signal faulty ASes to other ASes (e.g.
+core ASes or ASes in other ISDs).
+* ASes need adaptive becaon analysis algorithms that allow excluding specific beacons, e.g. beacons that have
+  known malicious ASes on path.
+* For swift recovery, it would be useful if ASes could revoke faulty segments.
+* Adapt the path discovery and dissemination algorithm to allow ASes to have many more than five paths to
```

Comment:
As a comment: This is essentially what the book promises, that the endpoints will have anyway a big selection to choose paths from.

Reply:
Done.